integrations/gitlab/gitlab_integration/git_integration.py

```diff
@@ -1,47 +1,125 @@
+    @staticmethod
+    def _validate_project_scope(data: Dict[str, Any]) -> Tuple[int, str]:
         if (project_id := data.get("id")) and (ref := data.get("default_branch")):
             return project_id, ref
         raise ValueError("Project id and ref are required")
 
+
+class SearchEntityProcessor(JQEntityProcessor):
+    prefix = SEARCH_PROPERTY_PREFIX
+    separation_symbol = "&&"
+
     def _search(self, data: Dict[str, Any], pattern: str) -> Any:
-        if pattern.startswith(FILE_PROPERTY_PREFIX):
-            project_id, ref = self._validate_project_scope(data)
-            logger.info(
-                f"Searching for file {pattern} in Project {project_id}, ref {ref}"
-            )
-            gitlab_service: GitlabService = event.attributes["project_id_to_service"][
-                project_id
-            ]
+        """
+        Handles entity mapping for search:// pattern
+        :param data: project data
+        :param pattern: e.g. search://scope=blobs&&query=filename:port.yml
+        :return: True if the search pattern matches, False otherwise
+        """
+        project_id, _ = self._validate_project_scope(data)
+        scope, query = self._parse_search_pattern(pattern)
+
+        # relying on that the code is being called after event initialization and as part of the GitLab service
+        # initialization
+        project: Project | None = event.attributes[PROJECTS_CACHE_KEY][project_id]
+        logger.info(f"Searching for {query} in Project {project_id}, scope {scope}")
+
+        match = False
+        if project:
+            match = bool(project.search(scope=scope, search=query))
+        return match
 
-            file_path = pattern.replace(FILE_PROPERTY_PREFIX, "")
-            return self._get_file_content(gitlab_service, project_id, file_path, ref)
+    def _parse_search_pattern(self, pattern: str) -> Tuple[str, str]:
+        """
+        :param pattern: e.g. search://scope=blobs&&query=filename:port.yml
+        :return: scope, search_pattern
+        """
+        # remove prefix
+        pattern = pattern.replace(self.prefix, "")
+        if len(pattern.split(self.separation_symbol)) == 1:
+            raise ValueError(
+                f"Search pattern {pattern} does not contain separation symbol {self.separation_symbol}"
+            )
+        # handle case when the query contains the separation symbol
+        elif len(pattern.split(self.separation_symbol)) > 2:
+            splitted_patten = pattern.split(self.separation_symbol)
+            scope = splitted_patten[0]
+            query = self.separation_symbol.join(splitted_patten[1:])
+        # default case
         else:
-            return super()._search(data, pattern)
+            scope, query = pattern.split(self.separation_symbol)
+
+        # return the actual scope and query values
+        # e.g. scope=blobs -> blobs, query=filename:port.yml -> filename:port.yml
+        return scope.split("=")[1], "=".join(query.split("=")[1:])
```

Comment:
If I understand this correctly, for checks with multiple rules, the query will end up being a `=`-delimited list, is that what's supposed to happen? Can you add an example of an expected query output for a case with multiple conditions?

Reply:
I join the list back to string after ignoring the first index which is the `query=` part of the string

Reply:
How does the list look after the second join? how will it look specifically is there is more than one search clause? (eg. `query=filename:port.yml&&filecontains:requests` - I just made that up but I want to see that it makes sense)

Reply:
So lets take for example the following: `query=pandas==2.6.0 filename:requirements.txt` which is the way you combine two conditions. query all requirements.txt files that contains `pandas==2.6.0` string. 

the output will be:
`pandas==2.6.0 filename:requirements.txt`

Reply:
Ok got it 👍